### PR TITLE
Scrum 9 cb 201 interactive feed

### PR DESCRIPTION
**Pull request by Devayani1612**
### Summary

Implemented the interactive feed UI for campus portal.

### Features Implemented

* Display posts with club_name, content, and images
* Like and Comment buttons with optimistic UI updates
* Scrollable middle feed section while keeping header and sidebars fixed
* Added header with logo for branding

### Technical Details

* Component-based structure using PostCard
* State management using React hooks
* Optimized layout using flex/grid with min-h-0 for proper scrolling

### Next Steps

* Integrate backend APIs for likes, comments, and posts
* Add validations and error handling

### CODE REVIEW

File: client/src/components/PostCard.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          alt="post"
+          className="rounded-lg mb-3 w-full max-h-80 object-cover"
+        />
+      )}
+
+      <div className="flex items-center gap-4 border-t border-slate-100 pt-3">
+        <button
+          onClick={() => onLike(post.id)}
+          className="text-sm font-medium text-slate-600 hover:text-blue-600"
+        >
```
Comment (Supraja050202) [inline client/src/components/PostCard.jsx:33]: These buttons look like plain text and lack clear affordance. Consider adding padding, background hover, and flex alignment so they feel more interactive and clickable

File: client/src/components/PostCard.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          onClick={() => onToggleComments(post.id)}
+          className="text-sm font-medium text-slate-600 hover:text-blue-600"
+        >
+          💬 Comment ({post.comments.length})
+        </button>
+      </div>
+
+      {post.showComments && (
+        <div className="mt-4 border-t border-slate-100 pt-3">
+          <div className="space-y-2 mb-3">
```
Comment (Supraja050202) [inline client/src/components/PostCard.jsx]: The spacing between comments feels cramped, making them harder to scan. Increasing vertical spacing (e.g., space-y-3) and adding subtle elevation (shadow) can improve readability and visual separation.

**Review by Supraja050202 — CHANGES_REQUESTED** [review]
The UI is clean and well-structured, but some interactive elements need better visual affordance.
Like and Comment buttons currently look like plain text and should feel more clickable with improved styling.
The comments section feels slightly cramped, which affects readability and user experience.
Increasing spacing and adding subtle visual separation would make the feed easier to scan.
Overall, small UI refinements will significantly improve usability and polish.

**Comment by Devayani1612** [comment]
Thanks for the feedback!

* Updated Like and Comment buttons with improved styling, padding, and hover states to enhance interactivity.
* Increased spacing and added subtle elevation to comments for better readability and visual separation.

Please review the changes.

**Review by Supraja050202 — DISMISSED** [review]
Looks great—these changes significantly improve the UI polish and usability.
The buttons now have clear affordance and feel interactive, and the hover states add nice feedback.
The increased spacing and subtle elevation in the comments section make it much easier to read and scan.
Everything aligns well with good UI/UX practices and keeps the design consistent.
Approved, good to merge!

File: client/src/components/PostCard.jsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+      { }
+      <p className="text-slate-700 mb-3">{post.content}</p>
+
+      { }
+      {post.image && (
+        <img
+          src={post.image}
+          alt="post"
+          className="rounded-lg mb-3 w-full max-h-80 object-cover"
```
Comment (premkiran2) [inline client/src/components/PostCard.jsx:24]: Nice layout for the action row the spacing and visual hierarchy make the Like/Comment CTA area easy to scan.

File: client/src/App.jsx [hunk 4]
```diff
@@ -1,59 +1,196 @@
...
+              
+              { }
               <div className="flex-1 overflow-y-auto p-4 space-y-4">
-                <div className="h-28 rounded bg-slate-200/80" />
-                <div className="h-28 rounded bg-slate-200/80" />
-                <div className="h-28 rounded bg-slate-200/80" />
-                <div className="h-28 rounded bg-slate-200/80" />
-                <div className="h-28 rounded bg-slate-200/80" />
-                <div className="h-28 rounded bg-slate-200/80" />
+                {posts.map((post) => (
```
Comment (premkiran2) [inline client/src/App.jsx:175]: The feed mapping reads really cleanly here, props are passed explicitly, and the component boundary is well-defined.

**Review by premkiran2 — DISMISSED** [review]
Approved! nice work! I reviewed the feed components and the optimistic UI flow and it’s coming together cleanly

**Review by Jahnavidarisetti — CHANGES_REQUESTED** [review]
Nice work on the interactive feed UI. Before merging, please add test cases to cover post rendering, optimistic like/comment interactions, and key edge cases like empty or large feeds to ensure stability as backend integration is added.

**Review by premkiran2 — APPROVED** [review]

**Review by Jahnavidarisetti — APPROVED** [review]
I think these test cases are good for now
